src/scrubber.ts

```diff
@@ -160,8 +161,13 @@ export class Scrubber {
     if (!a || !b) return false
     if (a === b) return true
 
+    // Remove any entries that are integers as we assume they are array indices that should be ignored for parent matching
+    let aSliced = a.filter(e => !e.match(/^[0-9]*$/g))
+
+    if (a.length < b.length) return false
+
     // a may be longer than b, slice a to the size of b, take chunk from the end
-    const aSliced = a.slice(a.length - b.length, a.length)
+    aSliced = aSliced.slice(aSliced.length - b.length, aSliced.length)
```

Comment:
After you put the length check above higher - you can combine `aSliced` operations and make it const: `const aSliced = a.filter(...).slice(...)`

Or, maybe not, as you rely on `.length`. Nevermind then.